internal/runtime/compiler/parser/parser.y

```diff
@@ -142,6 +149,27 @@ stmt
   }
   ;
 
+log_mapping_spec
+  : LOGMAPPING log_mapping_list NL
+  {
+    $$ = &ast.LogMapping{
+      P:        tokenpos(mtaillex),
+      Mappings: $2, // $2 is the list of strings (type []string)
+    }
+  }
+  ;
+
+log_mapping_list
+  : STRING
+  {
+    $$ = []string{$1}
+  }
+  | log_mapping_list COMMA STRING
```

Comment:
Turn this into a list parse like `metric_by_expr` where it has a terminating `id_or_string` and a recursing `log_filter_list, COMMA id_or_string`.